packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/useHandleUpload.ts

```diff
@@ -95,17 +107,25 @@ const mergeSelectedTasks = (
 export function useHandleUpload({
   prefix,
   preventOverwrite,
+  batchSize = 6,
 }: {
   prefix: string;
   preventOverwrite: boolean;
+  batchSize?: number;
 }): [
   tasks: CancelableTask[],
   handleUpload: () => void,
   handleFileSelect: (files: File[]) => void,
+  handleCancel: () => void,
 ] {
   const getConfig = useGetLocationConfig();
 
   const [tasks, setTasks] = React.useState<CancelableTask[]>(() => []);
+  const tasksRef = React.useRef<CancelableTask[]>(tasks);
+
+  React.useEffect(() => {
+    tasksRef.current = tasks;
+  }, [tasks]);
```

Comment:
for the life of me, I could not find any other way to get the inline cancel to work without this ref to the `tasks` array. Since we need to be able to cancel inline and also cancel all the tasks, I needed the most up to date status for each of the tasks and so the ref here was able to help with that

any feedback on this is appreciated

Reply:
What does storing `tasks` in a ref provide that usage of `tasks` directly does not?

Reply:
It helps gives access to the most up to date state of the tasks. This helps with knowing when to to skip over a task that was canceled

Previously when I tried to access `tasks` directly, it was always showing the initial state when we first start the uploads

Reply:
Believe that is because of the `Promise.all`?

Reply:
I think I was running into this issue when trying out the original method we discussed too. Since `setTasks` is react state, is it supposed to happen instantaneous?

By instantaneous, I wanted to be able to access the updated state within the next recursive call